ha/monitor/k8s/parser.py

```diff
@@ -114,6 +114,8 @@ def parse(self, an_event, cached_state):
             alert.event_type = an_event[K8SEventsConst.TYPE]
         if K8SEventsConst.NODE_NAME in an_event[K8SEventsConst.RAW_OBJECT][K8SEventsConst.SPEC]:
             alert.node = an_event[K8SEventsConst.RAW_OBJECT][K8SEventsConst.SPEC][K8SEventsConst.NODE_NAME]
+        if K8SEventsConst.NAME in an_event[K8SEventsConst.RAW_OBJECT][K8SEventsConst.METADATA]:
```

Comment:
I think you can store the value of an_event[K8SEventsConst.RAW_OBJECT][K8SEventsConst.METADATA] in a variable and use wherever required. Or I see RAW_OBJECT is common at all place. So store till that and use that.

Reply:
Yes you are right, But making changes in whole file will add more complexity to current task. In another case if we make changes only for this parameter (generation_id) then it will make it different from other parameter which is not acceptable. As it is parser, I think current changes are more readable instead making new variable & use it. Please let me know if you want me to make changes in whole file as scope of this ticket.

Reply:
Ok. Np. Just use that variable for your check only for now. (For line number 117 and 118).
Will that work?
If still you think it will be complicated, you can skip it then

Reply:
Made changes as per comment & Tested
```2021-12-22 11:13:07 pod_monitor [521]: INFO [__init__] Init done for pod monitor
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'host', '_resource_name': 'ssc-vm-g2-rhev4-2631.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_generation_id': None, '_node': None, '_is_status': True, '_timestamp': '1640171587'}
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'host', '_resource_name': 'ssc-vm-g2-rhev4-2632.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_generation_id': None, '_node': None, '_is_status': True, '_timestamp': '1640171587'}
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'node', '_resource_name': 'f5fca0c9f79c4ea8af90d75a707bb2ab', '_event_type': 'online', '_k8s_container': None, '_generation_id': 'cortx-data-pod-ssc-vm-g2-rhev4-2631-668fd5d96c-hzd5s', '_node': 'ssc-vm-g2-rhev4-2631.colo.seagate.com', '_is_status': True, '_timestamp': '1640171587'}
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'host', '_resource_name': 'ssc-vm-g2-rhev4-2633.colo.seagate.com', '_event_type': 'online', '_k8s_container': None, '_generation_id': None, '_node': None, '_is_status': True, '_timestamp': '1640171587'}
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'node', '_resource_name': '02b32e8117db49d1943fcd46c2a2c301', '_event_type': 'online', '_k8s_container': None, '_generation_id': 'cortx-data-pod-ssc-vm-g2-rhev4-2632-86846fb4fb-fpt7p', '_node': 'ssc-vm-g2-rhev4-2632.colo.seagate.com', '_is_status': True, '_timestamp': '1640171587'}
2021-12-22 11:13:07 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'node', '_resource_name': '533166337a5d4228b7da03de9b9b2302', '_event_type': 'online', '_k8s_container': None, '_generation_id': 'cortx-data-pod-ssc-vm-g2-rhev4-2633-7bf5986977-5x44p', '_node': 'ssc-vm-g2-rhev4-2633.colo.seagate.com', '_is_status': True, '_timestamp': '1640171587'}

After deletion of pod (cortx-data-pod-ssc-vm-g2-rhev4-2633-7bf5986977-5x44p) Logs observed are

2021-12-22 11:18:18 pod_monitor [521]: WARNING [parse] Exception received during parsing 'conditions'
2021-12-22 11:18:42 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'node', '_resource_name': '533166337a5d4228b7da03de9b9b2302', '_event_type': 'online', '_k8s_container': None, '_generation_id': 'cortx-data-pod-ssc-vm-g2-rhev4-2633-7bf5986977-pzx4g', '_node': 'ssc-vm-g2-rhev4-2633.colo.seagate.com', '_is_status': False, '_timestamp': '1640171922'}
2021-12-22 11:18:56 pod_monitor [521]: INFO [run] Sending alert on message bus {'_resource_type': 'node', '_resource_name': '533166337a5d4228b7da03de9b9b2302', '_event_type': 'failed', '_k8s_container': None, '_generation_id': 'cortx-data-pod-ssc-vm-g2-rhev4-2633-7bf5986977-5x44p', '_node': 'ssc-vm-g2-rhev4-2633.colo.seagate.com', '_is_status': False, '_timestamp': '1640171936'}
```

Reply:
Thanks!